DQMServices/Demo/test/test_dqmgui_upload.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/env bash
+# Simple test to use one of the files generated by TestDQMServicesDemo test
+# to upload it to the dev DQMGUI, making sure that the ROOT file format generated
+# is compatible. See cmssw issue #43590.
+# Requires curl and jq to be installed. Also requires either a Grid key and certificate,
+# or a proxy to be available.
+set -x
+set -e
+
+DEV_DQMGUI_URL="https://cmsweb.cern.ch/dqm/dev"
+# Create a unique fake "Era", so that different executions of this test
+# produce differently named files. This is required, because this test might pass
+# if it's checking a file that was successfully uploaded during a previous
+# instance of the same test.
+UNIQUE_ERA_ID=$(date -u +%Y%m%d%M%S%N)
+OLD_FILENAME=DQM_V0001_R000000001__Harvesting__DQMTests__DQMIO.root
+NEW_FILENAME=${OLD_FILENAME/DQMTests/DQMTests${UNIQUE_ERA_ID}}
+
+# Make sure the required file exists. We expect to find it in the current directory.
+if [ ! -f $OLD_FILENAME ]; then
+    echo "Unable to find required file ${OLD_FILENAME}!"
+    exit 1
+fi
+
+# Prepare curl args
+curl_args=""
+if [ -n "$X509_CERT_DIR" ]; then
+    curl_args="--capath ${X509_CERT_DIR}/ --key ${X509_USER_PROXY} --cert ${X509_USER_PROXY}"
+elif [ -n "$X509_USER_CERT" ] && [ -n "$X509_USER_CERT" ]; then
+    curl_args="--capath ${X509_CERT_DIR}/ --key ${X509_USER_KEY} --cert ${X509_USER_CERT}"
+elif [ -f $HOME/.globus/usercert.pem ] && [ -f $HOME/.globus/userkey.pem ]; then
+    curl_args="-k --key $HOME/.globus/userkey.pem --cert $HOME/.globus/usercert.pem"
+else
+    echo "Unable to find proxy or key/cert env vars, cannot continue"
+    exit 1
+fi
+
+# Make a copy with a filename that conforms to the naming convention required by DQMGUI.
+rm -f "$NEW_FILENAME"
+cp $OLD_FILENAME "$NEW_FILENAME"
+
+# Upload the file to the DQMGUI
+visDQMUpload.py "$DEV_DQMGUI_URL" "$NEW_FILENAME"
```

Comment:
For my education: will these files be indeed and stored permanently? Do we risk to eventually run out of db space if this is going to run twice a day times N IBs flavours for the foreseable future? Also do we risk to take down a production GUI with some kind of DDoS if some user inadvertenly runs this test multiple times? Sorry for the naive questions!

Reply:
> will these files be indexed and stored permanently?
Yes.

> eventually run out of db space if this is going to run twice a day times N IBs flavours for the foreseable future?
Yes, but this happens anyway for the dev flavor, since all CMSSW automated tests upload there all their files, which is why we clean it up every now and then. AFAIK no one will check for the tests of very old CMSSW PRs. 

> do we risk to take down a production GUI with some kind of DDoS
This test (same as all tests by cmsbuild) is uploading to the `dev` DQMGUI only (which is not "production" per se). Is your question about possibly bypassing the test to upload, e.g., to `offline` DQMGUI?

Reply:
> which is not "production" per se

If it's used for the PR tests, it's "production" in some sense. But ok if you already clean it up regularly I think it's fine,  thanks for the explanation.